src/ClassApplication/Controllers/ClassController.cs

```diff
@@ -1,121 +1,260 @@
-        ///</summary>
-        ///<param name="classId">Id of the classroom</param>
-        ///<param name="noteName">Name of the note to search for</param>
-        [HttpGet("{classId}")]
-        public async Task<ActionResult<string>> QueryNote(string classId, string noteName)
-        {
-            throw new NotImplementedException();
-        }
-    }
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using ACMTTU.NoteSharing.Shared.SDK.Controllers;
+using Microsoft.Azure.Cosmos;
+using ACMTTU.NoteSharing.Platform.ClassApplication.Services;
+using ClassApplication.Models;
+using System.Collections.Generic;
+
+namespace ACMTTU.NoteSharing.Platform.ClassApplication.Controllers
+{
+    [Route("api/classroom")]
+    [ApiController]
+    public class ClassController : PlatformBaseController
+    {
+        private Container classesContainer;
+        private PartitionKey partitionKey = new PartitionKey("classroom");
+
+        public ClassController(IHttpClientFactory factory, DBService databaseService) : base(factory)
+        {
+
+            // IF YOU'RE READING THIS: the _controller field should be PRIVATE,
+            //      but this programmer didn't have the clearance to abstract
+            //      the database functions in this controller.
+            //      please make this field PRIVATE upon abstracting the database
+            //      service from this class.
+            this.classesContainer = databaseService._container;
+        }
+
+        /// <summary>
+        /// 
+        /// Creates a classroom in the database
+        /// 
+        /// </summary>
+        /// <param name="ownerId">ID of the user creating the class</param>
+        /// <param name="classroomName">Name of the classroom being created</param>
+        /// <param name="description">String to set the new classroom description to (Optional)</param>
+        [HttpPost("CreateClass")]
+        public async Task<ActionResult<string>> CreateClassroom(string ownerId, string classroomName, string description = "")
+        {
+            string classId = Guid.NewGuid().ToString();
+            Classroom classroom = new Classroom(classId, ownerId, classroomName, description);
+            classroom.AddStudent(ownerId);
+
+            await classesContainer.CreateItemAsync<Classroom>(classroom, new PartitionKey(classId));
+
+            return Ok();
+        }
+
+
+
+        /// <summary>
+        /// 
+        /// Returns data from a classroom from a given classroom ID
+        ///
+        /// </summary>
+        /// <param name="classId">classroom id</param>
+        /// <returns>An array containing a value determined by the parameter</returns>
+        [HttpGet("GetClassByID/{classId}")]
+        public async Task<Classroom> GetClassroom(string classId)
+        {
+            Classroom classroom;
+            classroom = await classesContainer.ReadItemAsync<Classroom>(classId, new PartitionKey(classId));
```

Comment:
This doesn’t fail when the ID doesn’t exist? I know one of the read functions throws an exception and one just returns null.